### ISSUE TITLE
Poller is upgrading from pre Cacti 1.0, exiting till upgraded

### ISSUE THREAD

**Issue by eschoeller** [Cacti/cacti#4167]
This happened on one of my remote pollers. Three places where I suddenly started seeing some problems:

cacti.log:
```
03/13/2021 19:09:01 - POLLER: Poller[2] Poller is upgrading from pre Cacti 1.0, exiting till upgraded.
03/13/2021 19:10:01 - POLLER: Poller[2] Poller is upgrading from pre Cacti 1.0, exiting till upgraded.
03/13/2021 19:11:02 - POLLER: Poller[2] Poller is upgrading from pre Cacti 1.0, exiting till upgraded.
03/13/2021 19:12:01 - POLLER: Poller[2] Poller is upgrading from pre Cacti 1.0, exiting till upgraded.
03/13/2021 19:13:01 - POLLER: Poller[2] Poller is upgrading from pre Cacti 1.0, exiting till upgraded.
```
poller stderr output (and this was NOT caught into the stderr.log, sadly)
```
PHP Notice:  Undefined index: localhost:3306:cacti in /cacti/cacti-prod/lib/database.php on line 785
```

and the mysql error log:
```
2021-03-13  0:23:42 15848909 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  0:53:42 15857190 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  1:23:42 15865471 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  1:53:43 15873752 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  2:23:44 15882033 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  2:54:43 15890590 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  3:25:43 15899147 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  3:55:44 15907428 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  4:25:44 15915709 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  4:56:42 15924262 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  5:26:43 15932540 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  5:57:46 15941088 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  6:28:45 15949636 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  6:59:43 15958189 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  7:30:40 15966740 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  8:01:37 15975292 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  8:31:39 15983572 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  9:01:41 15991853 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13  9:32:42 16000410 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 10:02:43 16008691 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 10:33:41 16017248 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 11:03:42 16025528 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 11:34:40 16034083 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 12:04:42 16042361 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 12:35:41 16050914 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 13:05:43 16059194 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 13:36:42 16067751 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 14:07:43 16076308 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 14:38:41 16084864 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 15:08:41 16093144 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 15:38:43 16101424 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
2021-03-13 16:08:48 16109705 [Warning] InnoDB: Cannot add field `monitor_text` in table `cacti`.`host` because after adding it, the row size is 8830 which is greater than maximum allowed size (8126 bytes) for a record on index leaf page.
corrupted double-linked list
210313 16:09:50 [ERROR] mysqld got signal 6 ;
This could be because you hit a bug. It is also possible that this binary
or one of the libraries it was linked against is corrupt, improperly built,
or misconfigured. This error can also be caused by malfunctioning hardware.

To report this bug, see https://mariadb.com/kb/en/reporting-bugs

We will try our best to scrape up some info that will hopefully help
diagnose the problem, but since we have already crashed, 
something is definitely wrong and this may fail.

Server version: 10.3.27-MariaDB-0+deb10u1
key_buffer_size=16777216
read_buffer_size=131072
max_used_connections=56
max_threads=1026
thread_count=15
It is possible that mysqld could use up to 
key_buffer_size + (read_buffer_size + sort_buffer_size)*max_threads = 2272054 K  bytes of memory
Hope that's ok; if not, decrease some variables in the equation.

Thread pointer: 0x0
Attempting backtrace. You can use the following information to find out
where mysqld died. If you see no messages after this, something went
terribly wrong...
stack_bottom = 0x0 thread_stack 0x30000
```

**Comment by eschoeller** [comment]
mysql would not stop on its own, I had to kill -9 it.  upon restart:
```
2021-03-13 19:54:47 0 [Note] InnoDB: Using Linux native AIO
2021-03-13 19:54:47 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2021-03-13 19:54:47 0 [Note] InnoDB: Uses event mutexes
2021-03-13 19:54:47 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2021-03-13 19:54:47 0 [Note] InnoDB: Number of pools: 1
2021-03-13 19:54:47 0 [Note] InnoDB: Using SSE2 crc32 instructions
2021-03-13 19:54:47 0 [Note] InnoDB: Initializing buffer pool, total size = 6G, instances = 64, chunk size = 96M
2021-03-13 19:54:47 0 [Note] InnoDB: Completed initialization of buffer pool
2021-03-13 19:54:47 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2021-03-13 19:54:47 0 [Note] InnoDB: Starting crash recovery from checkpoint LSN=9780919361309
2021-03-13 19:54:48 0 [Note] InnoDB: Starting final batch to recover 621 pages from redo log.
2021-03-13 19:54:48 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
2021-03-13 19:54:48 0 [Note] InnoDB: Removed temporary tablespace data file: "ibtmp1"
2021-03-13 19:54:48 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2021-03-13 19:54:48 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2021-03-13 19:54:48 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2021-03-13 19:54:48 0 [Note] InnoDB: Waiting for purge to start
2021-03-13 19:54:48 0 [Note] InnoDB: 10.3.27 started; log sequence number 9780919388776; transaction id 4149866536
2021-03-13 19:54:48 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2021-03-13 19:54:48 0 [Note] Plugin 'FEEDBACK' is disabled.
2021-03-13 19:54:48 0 [Note] Recovering after a crash using tc.log
2021-03-13 19:54:48 0 [Note] Starting crash recovery...
2021-03-13 19:54:48 0 [Note] Crash recovery finished.
2021-03-13 19:54:48 0 [Note] Server socket created on IP: '192.168.252.2'.
2021-03-13 19:54:48 0 [Note] Reading of all Master_info entries succeeded
2021-03-13 19:54:48 0 [Note] Added new Master_info '' to hash table
2021-03-13 19:54:48 0 [Note] /usr/sbin/mysqld: ready for connections.
Version: '10.3.27-MariaDB-0+deb10u1'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  Debian 10
2021-03-13 19:54:48 0 [Note] InnoDB: Buffer pool(s) load completed at 210313 19:54:48
```

things appear to be OK again. So, I suppose mysql hit some really nasty state. Overall, my only suggestion here would be to handle that situation slightly differently in Cacti.

**Comment by TheWitness** [comment]
@eschoeller, I've found some odd mariadb issues with the last few releases, granted I'm doing some pretty advanced things at the moment.  The latest was a 3x performance degregation in poller performance between 10.5.8 and 10.5.9.  it was so bad I had to lock the clients repo to 10.5.8.

If you don't mind though, I would like to make this one 3rd party and close due to it being an edge/support case and there being so many other things in the backlog.
